File: python/src/tests/t_sleep_nanos.py
#!/usr/bin/python3
#
# Id:
#
# We test a bit of the atrshmlog here.
#
# This is for the first starter, so only the basic things.

import sys
import atrshmlog

result = atrshmlog.attach()

n = 1000000

while n > 0:
    t1 = atrshmlog.gettime()
    atrshmlog.sleep_nanos(n)
    t2 = atrshmlog.gettime()

    print('sleep nanos : ' + str(n) + ' : ' +  str(t2 - t1) + ' : ')
    n = int(n / 2)

print (' ')

exit(0)

# end of test

File: perl/src/tests/t_sleep_nanos.pl
#!/usr/bin/perl
#
# $Id:$
#
# perl test driver

use Atrshmlog;

$result = Atrshmlog::attach();

print "sleep nanos  : $result : \n";

for ($n = 1000000; $n ; $n >>= 1 ) {
    $t1 = Atrshmlog::gettime();
    Atrshmlog::sleep_nanos($n);
    $t2 = Atrshmlog::gettime();

    print "sleep nanos  : $n : " . ($t2 - $t1 ) . " : \n";

}


print " \n";

exit (0);

# end of main
